src/uenv/lex.cpp

```diff
@@ -110,9 +110,12 @@ class lexer_impl {
                 ++stream_;
                 return;
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wpedantic"
             case 'a' ... 'z':
             case 'A' ... 'Z':
             case '0' ... '9':
+#pragma GCC diagnostic pop
```

Comment:
This ties the build to GCC. The push/pop can be guarded if needed, but I guess you're not planning on targeting anything but GCC ever?

Reply:
We will only compile with GCC, and this does make the code _much_ more readable.
So let's keep it.

Reply:
Yep, this is a pretty neat extension (first time I see it!).